BaseTools/Scripts/BinToPcd.py

```diff
@@ -45,17 +47,25 @@ def ValidateGuidName (Argument):
             Message = '{Argument} is not a valid GUID C name'.format (Argument = Argument)
             raise argparse.ArgumentTypeError (Message)
         return Argument
+    
+    def XdrPackBuffer (buffer):

```

Comment:
Just curious did you write this based on the RFC description or is it derived from an existing implementation?

Reply:
derived from the xdrlib implementation with a few changes / improvements.